### PR TITLE
[tune] Cancel Experiment via Client

### LINKED ISSUE #7720: [tune] Should include pandas in the requirements
I installed tune in a Python 3.7.6 environment via `pip install ray[tune]`. Importing it in a program fails with:

```
<ipython-input-1-4384eba11eb4> in <module>                                                                                                                            [31/44]
----> 1 from ray import tune                                                                                                                                                 
                                                                                                                                                                             
~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/__init__.py in <module>                                                                 
      1 from ray.tune.error import TuneError                                                                                                                                 
----> 2 from ray.tune.tune import run_experiments, run                                                                                                                       
      3 from ray.tune.experiment import Experiment                                                                                                                           
      4 from ray.tune.analysis import ExperimentAnalysis, Analysis                                                                                                           
      5 from ray.tune.stopper import Stopper                                                                                                                                 
                                                                                                                                                                             
~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/tune.py in <module>                                                                     
      4 from ray.tune.error import TuneError                                                                                                                                 
      5 from ray.tune.experiment import convert_to_experiment_list, Experiment                                                                                               
----> 6 from ray.tune.analysis import ExperimentAnalysis                                                                                                                     
      7 from ray.tune.suggest import BasicVariantGenerator                                                                                                                   
      8 from ray.tune.trial import Trial                                                                                                                                     
                                                                                                                                                                             
~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/analysis/__init__.py in <module>                                                        
----> 1 from ray.tune.analysis.experiment_analysis import ExperimentAnalysis, Analysis                                                                                       
      2                                                                                                                                                                      
      3 __all__ = ["ExperimentAnalysis", "Analysis"]                                                                                                                         
                                                                                                                                                                             
~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/analysis/experiment_analysis.py in <module>                                             
     12 from ray.tune.result import EXPR_PROGRESS_FILE, EXPR_PARAM_FILE,\                                                                                                    
     13     CONFIG_PREFIX, TRAINING_ITERATION                                                                                                                                
---> 14 from ray.tune.trial import Trial                                                                                                                                     
     15 from ray.tune.trainable import TrainableUtil                                                                                                                         
     16                                                                                                                                                                      
                                                                                                                                                                             
~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/trial.py in <module>                                                                    
     11 from ray.tune import TuneError       
     12 from ray.tune.checkpoint_manager import Checkpoint, CheckpointManager
---> 13 from ray.tune.durable_trainable import DurableTrainable
     14 from ray.tune.logger import pretty_print, UnifiedLogger
     15 # NOTE(rkn): We import ray.tune.registry here instead of importing the names we                                                                                     

~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/durable_trainable.py in <module>                                                       
      1 import os
      2
----> 3 from ray.tune.trainable import Trainable, TrainableUtil
      4 from ray.tune.syncer import get_cloud_sync_client
      5

~/Downloads/software/miniconda3/envs/sw/lib/python3.7/site-packages/ray/tune/trainable.py in <module>                                                               
      7 import os
      8 import pickle
----> 9 import pandas as pd
     10 from six import string_type
ModuleNotFoundError: No module named 'pandas'
```

This should be listed as a dependency so it installs along with the `pip install`, or it shouldn't be a hard dependency.

I see an earlier issue #5366 but that's closed.

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?
Expose an API to cancel the experiment.

## Related issue number

Closes #7720

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/tests/test_tune_server.py [hunk 1]
```diff
@@ -119,6 +120,22 @@ def testStopTrial(self):
         self.assertEqual(
             len([t for t in all_trials if t["status"] == Trial.RUNNING]), 0)
 
+    def testStopExperiment(self):
+        """Check if Stop Trial works."""
```
Comment (richardliaw) [inline python/ray/tune/tests/test_tune_server.py]:
```suggestion
        """Check if stop_experiment works."""
```

**Review by ijrsvt — APPROVED** [review]

File: python/setup.py [hunk 2]
```diff
@@ -77,7 +77,7 @@
     "debug": [],
     "dashboard": [],
     "serve": ["uvicorn", "pygments", "werkzeug", "flask", "pandas", "blist"],
-    "tune": ["tabulate", "tensorboardX"]
+    "tune": ["tabulate", "tensorboardX", "pandas]
```
Comment (richardliaw) [inline python/setup.py]:
```suggestion
    "tune": ["tabulate", "tensorboardX", "pandas"]
```

File: python/ray/tune/web_server.py [hunk 3]
```diff
@@ -137,17 +142,21 @@ def do_PUT(self):
...
-                result = self._get_trial_by_url(self.path)
                 resource = {}
-                if result:
-                    if isinstance(result, list):
-                        infos = [self._trial_info(t) for t in result]
-                        resource["trials"] = infos
-                        for t in result:
+
+                if self.path.endswith("stop_experiment"):
+                    print("requesting experiment stop")
```
Comment (richardliaw) [inline python/ray/tune/web_server.py]:
```suggestion
```

File: python/ray/tune/trial_runner.py [hunk 4]
```diff
@@ -688,7 +690,10 @@ def _update_trial_queue(self, blocking=False, timeout=600):
     def request_stop_trial(self, trial):
         self._stop_queue.append(trial)
 
-    def _process_requests(self):
+    def request_stop_experiment(self):
```
Comment (ujvl) [inline python/ray/tune/trial_runner.py]: nit: documentation explaining behavior (eg when the request will actually be processed)

**Review by ujvl — APPROVED** [review]

**Pull request merged**